### ISSUE TITLE
Add layers implementation for Object detection & Image Segmentation model at ONNX model Zoo

### ISSUE THREAD

**Issue by ashishkrshrivastava** [opencv/opencv#16749]
##### System information (version)

- OpenCV => 3.4.9-dev
- Operating System / Platform => Ubuntu

##### Detailed description

Layer implementation of few layers that are needed to load Object detection and Image segmentation model from ONNX model Zoo are not present. Some of them includes-
`Ceil`
`Floor`
`Log`
`Round `
`Sqrt`

**Few examples** - 
[ObjectDetection&ImageSegmentation/yolov3/yolov3](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/yolov3)
`Ceil`
[ObjectDetection&ImageSegmentation/faster_rcnn_R_50_FPN_1x/faster_rcnn_R_50_FPN_1x](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/faster-rcnn)
`Floor`
`Log`
`Sqrt`

##### Steps to reproduce
Not an error but feature request.

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution

 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by dkurt** [comment]
Issue seems incomplete. Can you refer which exact object detection typologies are mentioned?

**Comment by ashishkrshrivastava** [comment]
added.

**Comment by dkurt** [comment]
> ObjectDetection&ImageSegmentation/yolov3/yolov3
Ceil

For this YOLOv3 adding Ceil layer won't solve the problem. There are also Loop, Tile, NonMaxSupression etc. Implementing them separately is not a good solution as for me - it should be origin Reorg and Region layers.

The same story with Faster R-CNN

Can you please do some research and provide what is the origin of this ONNX models? Have they been trained in PyTorch or just converted from TensorFlow models?

**Comment by ashishkrshrivastava** [comment]
```
And
Cast
Ceil 
ConstantOfShape
Exp 
Expand
Equal
Floor 
Greater
Less
Log 
Loop
NonMaxSuppression
Not 
ReduceMin
RoiAlign
Round 
Sqrt 
Tile
TopK

```
These are all of the layers that need implementation in order to run all ONNX object detection and image segmentation models. 
PR #16750 adds `Ceil, Exp, Floor, Log, Not, Round, Sqrt`
`And, Equal, Greater, Less` can be easily implemented using `scale` layer. 
Remaining layers like` Cast, ConstantofShape, Expand, Loop, NonMaxSupression, ReduceMIn, RoiAlign, Tile and TopK` needs separate implementation. 

Actually, First I started with finding pattern and fuse that to layers like YOLO as specified in YOLOv3.cfg file from darknet. But that network is really messy and finding pattern was not easy, Faster-RCNN was even bigger and some others were even bigger. So fusion for all of them won't be easy since each of them have different patterns. 
Yeah, this is right that whatever the model be and how much complex it looks like but they must have unfused subgraphs like `Origin, Reorg, Region` and others. So I tried to find repeated pattern for` yolov3` but along with repeated pattern there are some extra layers, some arbitary combinations of layers. The models are originated from Keras, so the implementation might not be same as we see in the model implementation in Darknet cfg files. Same for other networks. 
Then I compiled all the layers that needed implementation and found that there are only 20 layer that needs implementation to run all Object detection and segmentation models in which 11 can easily implemented from `eltwise` and `scale `layers. 

In future we could see some others networks with different layer combinations and patters, but fundamentally they cannot have operators other than ONNX defined operators (that are nearly 150, most of them are mathematical so could be easily implemented using some predefined layers like `eltwise, scale, elementwise`, etc)  so if we implement all of the needed operators of ONNX then we need to care about model implementation. 

This was my idea. Please share your view on this.

**Comment by dkurt** [comment]
> The models are originated from Keras

So this is a key thing. Why we need to convert Keras model to ONNX to import then in OpenCV? For TensorFlow we already have approach with text graphs described in https://github.com/opencv/opencv/wiki/TensorFlow-Object-Detection-API. It helped to cover SSD, Faster R-CNN and Mask R-CNN topologies.

**Comment by ashishkrshrivastava** [comment]
ohk, so we need not look after ONNX model zoo models that are originated from keras ? Since all of the object detection and segmentation models are originated from keras, should we leave all of them ?

And what do you think about layer implementation ? should I focus on subgraph fusion or layer implementation in future ?

**Comment by dkurt** [comment]
@ashishkrshrivastava, we can take a look at models from PyTorch in example. As for me, we need to enable networks in a single PR. So if we introduce some simple layer first but then could not enable something like `Loop` - all other layers would be useless.

So my point is focus on model rather than layers/ops - we need to enable models.

**Comment by ashishkrshrivastava** [comment]
Got it, thanks.

**Comment by ashishkrshrivastava** [comment]
I think this issue isn't relevant anymore. Closing it for now.
